app/views/handlers/individual_response.py

```diff
@@ -826,19 +848,78 @@ def handle_get(self):
     def handle_post(self):
         if self.selected_option == self.confirm_option:
             self._update_section_status(CompletionStatus.INDIVIDUAL_RESPONSE_REQUESTED)
+
+            serialized_mobile_number = self._request_args.get("mobile_number")
+            self._publish_fulfilment_request(serialized_mobile_number)
             return redirect(
                 url_for(
-                    "individual_response.get_individual_response_text_message_confirmation",
+                    "individual_response.individual_response_text_message_confirmation",
                     journey=self._request_args.get("journey"),
-                    mobile_number=self._request_args.get("mobile_number"),
+                    mobile_number=serialized_mobile_number,
                 )
             )
 
         return redirect(
             url_for(
-                "individual_response.get_individual_response_text_message",
+                "individual_response.individual_response_text_message",
                 list_item_id=self._list_item_id,
                 journey=self._request_args.get("journey"),
                 mobile_number=self._request_args.get("mobile_number"),
             )
         )
+
+
+class FulfilmentRequest:
+    def __init__(self, metadata: Mapping, mobile_number: Optional[str] = None):
+        self._metadata = metadata
+        self._mobile_number = mobile_number
+        self._fulfilment_type = "sms" if self._mobile_number else "postal"
+
+    def _get_individual_case_id_mapping(self) -> Mapping:
+        return (
+            {}
+            if self._metadata.get("case_type") == "SPG"
+            else {"individualCaseId": str(uuid4())}
+        )
+
+    def _get_mobile_number_mapping(self) -> Mapping:
```

Comment:
This method is called to populate the `contact` property so would make more sense as:
```suggestion
    def _get_contact_mapping(self) -> Mapping:
```

Reply:
Updated all ocurrences.